src/lib/GUID.cpp

```diff
@@ -70,20 +122,72 @@ bool GUID::is_equal(GUID other)
 }
 
 // This is used for FileDataStores. The GUID is given as a string in the file
-void GUID::from_string(std::string str)
+void GUID::from_string(std::string const str)
 {
   ONE_DEBUG_MSG(("\n"));
 
-  Data1 = strtol(str.substr(0, 8).c_str(), NULL, 16);
-  Data2 = strtol(str.substr(9, 4).c_str(), NULL, 16);
-  Data3 = strtol(str.substr(14, 4).c_str(), NULL, 16);
-  Data4[0] = strtol(str.substr(19, 4).c_str(), NULL, 16);
-  Data4[1] = strtol(str.substr(24, 4).c_str(), NULL, 16);
-  Data4[2] = strtol(str.substr(28, 4).c_str(), NULL, 16);
-  Data4[3] = strtol(str.substr(32, 4).c_str(), NULL, 16);
+  if (str.size() < MinStringGUIDLength)
+  {
+    return;
+  }
+
+  if (str.front() == '{' && str.size() > MaxStringGUIDLength)
+  {
+    return;
+  }
+
+
+  size_t i{0};
```

Comment:
What is the meaning of `{0}` here? Does it make a difference somewhere later on?

Reply:
The `0` is not necessary here. I'll make the initialization implicit.